File: 2515-shortest-distance-to-target-string-in-a-circular-array/solution.cpp
#include <vector>
#include <string>
#include <algorithm>
#include <cmath>

using namespace std;

class Solution {
public:
    int closetTarget(vector<string>& words, string target, int startIndex) {
        int n = words.size();
        int minDistance = -1;

        for (int i = 0; i < n; ++i) {
            if (words[i] == target) {
                // Calculate the absolute distance between indices
                int directDist = abs(i - startIndex);
                
                // In a circular array of size n, the distance between two indices 
                // is either the direct distance or the wrap-around distance
                int currentDist = min(directDist, n - directDist);
                
                // Initialize minDistance or update if a smaller distance is found
                if (minDistance == -1 || currentDist < minDistance) {
                    minDistance = currentDist;
                }
            }
        }

        return minDistance;
    }
};

File: 2515-shortest-distance-to-target-string-in-a-circular-array/solution.py
class Solution:
    def closetTarget(self, words: list[str], target: str, startIndex: int) -> int:
        n = len(words)
        min_distance = float('inf')
        found = False
        
        for i in range(n):
            if words[i] == target:
                found = True
                # Direct distance between indices
                direct_dist = abs(i - startIndex)
                
                # The minimum distance in a circular array is the minimum of
                # the direct distance and the distance going the other way around
                current_dist = min(direct_dist, n - direct_dist)
                
                if current_dist < min_distance:
                    min_distance = current_dist
        
        return min_distance if found else -1